### ISSUE TITLE
Issue (with material ui) after changing the default debounceRendering in 10.10

### ISSUE THREAD

**Issue by krskrs** [preactjs/preact#3724]
Hi,

I've found another case where the legacy debounceRendering mode was 'more compatible' with React.

Please see codepen below:

https://codesandbox.io/s/adoring-firefly-n9fjg2?file=/src/index.js:322-339

now, the issue is with the Birthday label wrongly animating on component rendering (the animation should not be seen).

To reproduce, reload the codepen with its button until you see the animation (it should be easily visible). In case codepen shows some errors, just reload again, and they should disappear.

Then uncomment the option line in code, try reloading many times, and the animation should not be showing anymore, which is the correct behavior.

Now I'm a bit confused about this and would like to bring this matter to your attention. I've seen that this change has fixed some old issues, but at the same time, it looks like new issues are arising with the changed debounce method. This one, another one in your issue tracker about react-big-calendar drag'n drop.

**Comment by krskrs** [comment]
UPDATE: It looks like I can't apply the fix locally in my IDE. I tried to change the options starting from the index.ts down to the component where DatePicker is, but I couldn't stop the animation. On codepen, the 'fix' seems to work.

**Comment by krskrs** [comment]
UPDATE 2: on further investigation it looks like it can also be reproduce on REACT! What still remains weird is the codepen I had created which seemed to demonstrate it could have been related with the debounceRendering method....

**Comment by arpu** [comment]
we have this problem to on our preact three.js app since version 10.10

**Comment by JoviDeCroock** [comment]
@krskrs is it safe to close this for now?

**Comment by krskrs** [comment]
Hi @JoviDeCroock, I opened an issue on their repo ( https://github.com/mui/mui-x/issues/6140#issuecomment-1246891697 ) and they acknowledged the bug, which I was able to restrict to a specific reproducible case, also on React (a datetime picker with a textarea).

The only thing here which could be of interest to Preact is the example I posted in the CodeSandbox, where there seem to be a different behavior between the new and the legagy debounceRednering configuration. But it's also true that, weird enough, such behavior is reproducible in codesandbox, but not on my local machine!

So, I'd say I  leave it up to you, if you think it's a thing worth investigating as an additional example of the differences/side effects of the changed debounceRendering, or if it somehow is a spurious behavior caused by the CodeSandbox environment (even if the debounceRendering still seems to be the discriminant).
